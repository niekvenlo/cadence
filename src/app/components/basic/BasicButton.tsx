"use client";

import type { ReactNode } from "react";

type Props = {
  children: ReactNode;
  variant?: "primary" | "look-like-a-link";
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
};

export default function BasicButton({ children, variant, onClick }: Props) {
  return (
    <button data-variant={variant} onClick={onClick}>
      {children}
    </button>
  );
}
