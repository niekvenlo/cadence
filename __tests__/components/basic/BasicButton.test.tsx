import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import BasicButton from "../../../src/app/components/basic/BasicButton";

describe("BasicButton", () => {
  it("renders a basic button", () => {
    render(<BasicButton>label</BasicButton>);
    const button = screen.getByRole("button");

    expect(button).toBeInTheDocument();
    expect(button).toHaveTextContent("label");
    expect(button).not.toHaveAttribute("data-variant");
    expect(button).toMatchSnapshot();
  });
  it("renders a primary button", () => {
    render(<BasicButton variant="primary">label</BasicButton>);
    const button = screen.getByRole("button");

    expect(button).toBeInTheDocument();
    expect(button).toHaveTextContent("label");
    expect(button).toHaveAttribute("data-variant");
    expect(button).toMatchSnapshot();
  });
});
