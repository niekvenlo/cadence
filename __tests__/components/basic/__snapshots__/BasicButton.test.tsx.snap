// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicButton renders a basic button 1`] = `
<button>
  label
</button>
`;

exports[`BasicButton renders a primary button 1`] = `
<button
  data-variant="primary"
>
  label
</button>
`;
